resolver/mindeps/__main__.py

```diff
@@ -91,39 +92,43 @@ def main_parser() -> argparse.ArgumentParser:
     return parser
 
 
-def task() -> None:  # noqa: C901
-    parser = main_parser()
-    args = parser.parse_args()
+class MinimumDependencyProvider(resolver.Provider):
+    def sort_candidates(
+        self,
+        candidates: Iterable[resolver.Candidate],
+    ) -> Sequence[resolver.Candidate]:
+        return sorted(candidates, key=operator.attrgetter('version'), reverse=False)
```

Comment:
Is there any particular reason you moved this here? Unless there's a good reason not to, I think we should keep it in `resolver.mindeps`. Also notice that removing it from there breaks imports, this class is meant to be available for people to use.